src/Store/Reducers/reducersHelper.js

```diff
@@ -64,21 +64,29 @@ export const addOrRemoveItemFromObj = (targetObj, inputArr) => {
     return { ...targetObj, ...inputObj };
 };
 
-export const getNewSelectedItems = (selectedItems, currentItems) => {
-    let payload = [].concat(selectedItems).map(({ id, selected }) => ({ id, selected }));
-    return addOrRemoveItemFromObj(currentItems, payload);
-};
-
 export const selectRows = (state, action) => {
-    const selectedUpdated = getNewSelectedItems(action.payload, state.selectedRows);
-    let prevLoadedRows = [].concat(state.prevLoadedRows);
-    let selectedRowsRawData = prevLoadedRows.filter(({ id }) => selectedUpdated[id] === true);
+    let selectedRows = state.selectedRows.slice();
+    if (Array.isArray(action.payload)) {
+        // when the state and payload are the same length it means that it is a "select none" action
+        if (state.selectedRows.length && state.selectedRows.length === action.payload.length) {
```

Comment:
This approach is a bit problematic -- what if I have 20 items, navigate to other page and select whole page (20 items), then I have 0 items selected, but I should have 40. Is it possible to pass a boolean indicating whether we are selecting or deselecting into `selectRows` function? Then you could get rid of this line
`const selectedRow = selectedRows.find(selected => selected.id === action.payload.id);`
in the `else` clause as well.

Reply:
Yeah I thought about using a boolean too, even better would be to use a different action than `SELECT_ENTITY`, like `DESELECT_ENTITY` :D I don't know

Reply:
That's up to you, both ways would work; I'd prefer just boolean so we don't have too many actions.

Reply:
Updated to use boolean